### ISSUE TITLE
Failed test 'No tests run for subtest "check_isbns"'

### ISSUE THREAD

**Issue by eserte** [briandfoy/business-isbn-data#8]
The test suite fails on some of my smokers:
```
...
t/check_data_structure.t .. ok
    # No tests run!

#   Failed test 'No tests run for subtest "check_isbns"'
#   at t/default_data.t line 40.
# Looks like you failed 1 test of 4.
t/default_data.t .......... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/4 subtests 
    # Business::ISBN::Data 20210112.004
t/load.t .................. ok
...
```
This seems to happen if a version of Business::Data is already installed and thus the t/default_data.t test is not skipped.
Maybe related: the variable `@isbns_from_issues` in https://metacpan.org/source/BDFOY/Business-ISBN-Data-20210112.004/t%2Fdefault_data.t#L8 is nowhere used, instead in the check_isbns subtest there's a loop over `$_[0]` which isn't defined.

**Comment by briandfoy** [comment]
I pushed a new version that should fix this.

**Comment by briandfoy** [comment]
This is still failing, but for other issues that I can't ascertain. It only fails in certain CPAN Testers, and not in Github Actions or Appveyor. Very curious.

**Comment by eserte** [comment]
It fails if Business::ISBN 3.004 or older is already installed. Statistical analysis of reports from my smokers:
```
$ local-ctgetreports.pl Business-ISBN-Data-20210112.005_02  --q 'qr:Business::ISBN is (\S+)'
...
Regression 'qr:Business::ISBN is (\S+)'
****************************************************************
Name                   Theta          StdErr     T-stat
[0='const']           1.0000          0.0000    40559448850986032.00
[1='eq_3.003']       -1.0000          0.0000    -20806558246808380.00
[2='eq_3.004']       -1.0000          0.0000    -21540515077319708.00
[3='eq_3.005']       -0.0000          0.0000      -2.67
[4='eq_3.006']        0.0000          0.0000       2.96

R^2= 1.000, N= 79, K= 5
****************************************************************
```

Problem is: in this situation it is impossible to update both, Business::ISBN and Business::ISBN::Data. The latter fails (hence this report), the former depends on Business::ISBN::Data and cannot be upgraded.

**Comment by eserte** [comment]
Here's how the problem can be reproduced with the help of a docker container. Save the below as "Dockerfile" in an empty directory and run `docker build -t perl-test . && docker run -it perl-test`

```
FROM debian:10

RUN apt-get -y update
RUN apt-get -y --no-install-recommends install perl make cpanminus
RUN apt-get -y --no-install-recommends install curl ca-certificates
RUN cpanm --notest Business::ISBN@3.004
#CMD cpanm --verbose --test-only Business::ISBN::Data
#CMD cpanm https://cpan.metacpan.org/authors/id/B/BD/BDFOY/Business-ISBN-Data-20210112.005_03.tar.gz
CMD cpanm --verbose --test-only https://cpan.metacpan.org/authors/id/B/BD/BDFOY/Business-ISBN-Data-20210112.005_02.tar.gz
```

**Comment by briandfoy** [comment]
Thanks, as I was going to bed I thought I should ask for your statistical analysis and then looked a final time and saw the pattern. But I didn't post anything.

Looking back at Business::ISBN 3.004, I see there was a bug with handling the ISBN13 strings. This was fixed in 3.005. That's a pretty big change for a point release, which is partly why I ignored the signals.

Now everything passes. 20210112.006 is on its way to CPAN, so we'll see how CPAN Testers reacts.
